default_plugins/consoleme_default_plugins/plugins/metrics/cloudwatch/__init__.py

```diff
@@ -0,0 +1,61 @@
+import threading
+from typing import Dict, Optional, Union
+
+import boto3
+
+from consoleme.config import config
+
+cloudwatch = boto3.client("cloudwatch", region_name=config.region)
+
+
+class CloudWatchMetric:
+    def __init__(self):
+        self.namespace = config.get("metrics.cloudwatch.namespace", "ConsoleMe")
+
+    def send_cloudwatch_metric(self, metric_name, dimensions, unit, value):
+        cloudwatch.put_metric_data(
+            Namespace=self.namespace,
+            MetricData=[
+                {
+                    "MetricName": metric_name,
+                    "Dimensions": dimensions,
+                    "Unit": unit,
+                    "Value": value,
+                },
+            ],
+        )
+
+    def generate_dimensions(self, tags):
+        dimensions = []
+        if not tags:
+            return dimensions
+        for name, value in tags.items():
+            dimensions.append({"Name": str(name), "Value": str(value)})
+        return dimensions
+
+    def count(self, metric_name, tags=None):
+        dimensions = self.generate_dimensions(tags)
+
+        threading.Timer(
```

Comment:
Will this spawn a thread for each metric sent? Should we use a thread pool instead?

Reply:
Using threadpool now